### DISCUSSION TITLE
tutorials/intro-typescript/11-query-arguments

### DISCUSSION THREAD

**Discussion by giscus** [apollographql-education/content-feedback#175] [course feedback]
# tutorials/intro-typescript/11-query-arguments

https://www.apollographql.com/tutorials/intro-typescript/11-query-arguments

**Comment by lindjacob** [comment]
I am running into a problem after I update my scripts to:
```
"scripts": {
    "dev": "concurrently \"ts-node-dev --respawn --watch ./**/*.graphql ./src/index.ts\" \"npm run generate\",
    "generate": "graphql-codegen --watch \"src/schema/odyssey-guide/types.graphql\""
  }
```
When I run "npm run dev" in my CLI, the server never starts up.

However, if I remove the `--watch ./**/*.graphql` from the `dev` script and instead expand the watch parameter for the `generate` script to include all .graphql files, then my server will start and it reacts to changes as I would have expected:
```
"scripts": {
    "dev": "concurrently \"ts-node-dev --respawn ./src/index.ts\" \"npm run generate\"",
    "generate": "graphql-codegen --watch \"./**/*.graphql\""
}
```
However, I would like to know if I am the only one getting this error and if there might be any issues with this change.

**Reply by lindjacob** [reply]
I have found that my current setup doesn't allow my to edit in `resolvers.ts`. When doing so I get a compile error.

**Reply by lizhennessy** [reply]
Hi there @lindjacob ! Looks like you've got a different file system than the one we follow in the course, is that right? (For instance, your schema file is `types.graphql` inside of `src/schema/odyssey-guide`? ) If you have your code pushed up somewhere, I'd be happy to take a look! Also interested in the compile error you see when editing `resolvers.ts`?

**Reply by lindjacob** [reply]
Hi Liz

Thank you for getting back to me so quickly.

I am trying to narrow down what the issue could be and in my research I tested out your repo's branch called `final`.
During this test I encountered something new, that seemed off. When changing the id parameter name in `schema.graphql` and then after changing the parameter in the `resolvers.ts`, I get the following error:
```
[0] [ERROR] 09:51:08 ⨯ Unable to compile TypeScript:
[0] src/resolvers.ts(9,20): error TS2339: Property 'listingId' does not exist on type 'RequireFields<QueryListingArgs, "id">'.
```
If I restart the server by running the `npm run dev` twice, I first resolve the problem (recreate a new `types.ts` files) and second time the server starts.
I have also created a video showing this for you: https://streamable.com/k4b630

This is not what I would expect from a hot reloading setup.
I am wondering if you can reproduce this issue?

**Reply by lindjacob** [reply]
Hi Liz

I solved my problem by switching my node version from 22 to 20 in my container.
However, the problem described above persists. My guess is that the server attempts to start before the `types.ts` can be generated because the two processes are started concurrently. I would love to hear your input.

**Reply by lindjacob** [reply]
Haha nevermind actually. The problem is back. My colleague couldn't start the server and after I rebuilt my container I also couldn't start it again. However, the server starts locally. So the issue only exist in our docker environment.
We are still looking into it.

**Reply by lindjacob** [reply]
I believe we found a solution now. Changing the `--watch` to only look at subfolders and files of the `src/` folder rather than all folders and files in the repo - which includes the `node_modules/`.
We are working in a container which less resources. That could be the difference between running the `--watch` locally.
Our working script looks like this:
```
"scripts": {
    "compile": "tsc",
    "dev": "concurrently \"ts-node-dev --respawn --watch ./src/**/*.graphql ./src/server.ts\" \"npm run generate --watch\"",
    "start": "npm run compile && nodemon ./dist/server.js",
    "test": "jest",
    "generate": "graphql-codegen --watch \"src/schema/odyssey-guide/types.graphql\""
},
```
\
One more thing. 
There is a difference between your script in your github repo on the `final` branch and here in your guide.
In your repo you have `--watch` parameter at the end of your `"dev"` script:
```
...\"npm run generate --watch\""
```
But here in the guide there is no `--watch` parameter there in the `"dev"` script:
```
...\"npm run generate\""
```
Since there is already a `--watch` parameter inside our `"generate"` script:
```
"generate": "graphql-codegen --watch \"src/schema.graphql\""
```
does that make it redundant to also have it in our `"dev"` script calling the `"generate"` script?

**Comment by lindjacob** [comment]
I have another question. The current setup only watches for changes in a single .graphql file. This works fine for smaller projects, but if we were to use an exploded architecture for different sources and tables (like one person suggest here https://theehsansarshar.hashnode.dev/scalable-graphql-architecture), then we would need to watch multiple files or combine the different .graphql files into one.
What approach would you suggest and can you give a short example on how you would do it?

**Comment by proselin** [comment]
### Code challenge error
In the first practice code challenge, it required the field `galaxy` contain an argument id of type **non-nullable ID** and **return a nullable Galaxy**. I code exact what it require.But when i submit that code the website return error that **galaxy field** > **should return a non nullable Galaxy type**. Please check this

**Reply by lizhennessy** [reply]
Great catch @proselin !!! This has been fixed in the course. Please do let us know if you encounter anything else that looks amiss.

**Reply by Arriv9l** [reply]
Hi @lizhennessy , I found another one similar to this. Could you please take a look.
![image](https://github.com/user-attachments/assets/6614f0e1-dbd5-4a52-9509-c172421ceb97)

**Reply by lizhennessy** [reply]
Oop! Great callout @Arriv9l , I'm on it!
